### DISCUSSION TITLE
Any suggestion to speed up run_scenicplus

### DISCUSSION THREAD

**Discussion by yupingz** [aertslab/scenicplus#77] [Q&A]
My run_scenicplus step had run for over 3 days and still was not finished (I used 20 cores). This is the size of my data
SCENIC+ object with n_cells x n_genes = 2165 x 18113 and n_cells x n_regions = 2165 x 258529

Are the number of regions too high? Any suggestion how to speed up run_scenicplus?

**Comment by yupingz** [comment]
increasing --shm-size when start docker run and increasing n_cpu to as many as possible

**Comment by SeppeDeWinter** [comment]
Hi @yupingz 

Sorry, I missed this question before.

The numbers of cells, regions and genes are not too high (same order of magnitude in fact as the PBMC data used in the tutorial). Three days is very long ... Do you have any idea where the workflow gets stuck?

Again using the development code for the TF-to-gene step might help here https://github.com/aertslab/scenicplus/commit/ec1d8d0a6fc398f88bf64e94af4ae4b4280fe6c9.

Best,

Seppe

**Comment by yupingz** [comment]
Thank you. I forgot which step took the longest. I increased  --shm-size and cpu in a different run and it finished in a much short time.

**Reply by SeppeDeWinter** [reply]
Ok. That's great to hear!

Best,

Seppe
